sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TypeUtils.scala

```diff
@@ -53,4 +53,13 @@ object TypeUtils {
 
   def getOrdering(t: DataType): Ordering[Any] =
     t.asInstanceOf[AtomicType].ordering.asInstanceOf[Ordering[Any]]
+
+
+  def compareBinary(x: Array[Byte], y: Array[Byte]): Int = {
```

Comment:
this seems super slow ... after this PR we should create a ByteArrayUtils in unsafe.types